server/src/main/java/org/elasticsearch/monitor/fs/FsHealthService.java

```diff
@@ -109,42 +111,53 @@ public void setSlowPathLoggingThreshold(TimeValue slowPathLoggingThreshold) {
 
     @Override
     public StatusInfo getHealth() {
-        StatusInfo statusInfo;
-        Set<Path> unhealthyPaths = this.unhealthyPaths;
         if (enabled == false) {
-            statusInfo = new StatusInfo(HEALTHY, "health check disabled");
-        } else if (brokenLock) {
-            statusInfo = new StatusInfo(UNHEALTHY, "health check failed due to broken node lock");
-        } else if (unhealthyPaths == null) {
-            statusInfo = new StatusInfo(HEALTHY, "health check passed");
-        } else {
-            String info = "health check failed on ["
-                + unhealthyPaths.stream().map(k -> k.toString()).collect(Collectors.joining(","))
-                + "]";
-            statusInfo = new StatusInfo(UNHEALTHY, info);
+            return new StatusInfo(HEALTHY, "health check disabled");
```

Comment:
NIT: while we're at it, these constant `StatusInfo` returns could be constants I guess :)